lib/linguist/languages.yml

```diff
@@ -5134,6 +5134,16 @@ Sieve:
   codemirror_mode: sieve
   codemirror_mime_type: application/sieve
   language_id: 208976687
+Singularity:
+  type: programming
+  color: "#64E6AD"
+  tm_scope: source.singularity
+  extensions:
+  - ".def"
```

Comment:
> The `.def` extension doesn't conflict with another language within linguist, but I've added a heuristic as it seems to be a commonly used extension for arbitrary text and config files.

If we add it for Singularity, we will also need to add it for other languages using it. Otherwise, Linguist will associate all `.def` files to Singularity, even if there's a heuristic rule defined for it.

That is likely to be a significant job since a lot of other languages appear to be using it.